proposals/interface-types/working-notes/Types.md

```diff
@@ -0,0 +1,99 @@
+# Interface value types
+
+This proposal defines new interface values, which have corresponding interface value types. Interface value types are a superset of the [core value types](https://webassembly.github.io/spec/core/syntax/types.html#value-types)
+
+Some interface value types are compound by being constructed using one or more interface value types. The abstract syntax is designed to prevent recursive types by not allowing a compound type definition to refer to itself or other definitions. This is so that the set of control flow needed to generate all interface values is simple enough to allow for optimized lazy semantics.
+
+The interface value type syntax is summarized below. Following after that is a section for each type that gives the full syntax, possible values, validation rules, and subtyping relationships.
+
+```
+#interface-valtype ::=
+  | #valtype
+  | bool
+  | s8
+  | s16
+  | s32
+  | s64
+  | u8
+  | u16
+  | u32
+  | u64
+  | string
+  | record ..
+  | array ..
+```
+
+## `bool` 
+
+A boolean value. The set of `{true, false}`.
+
+## `s8`, `s16`, `s32`, `s64`, `u8`, `u16`, `u32`, `u64`
+
+Integer values.
+
+| Type | Values            |
+|------|-------------------|
+| s8   | [-(2^7), 2^7-1]   |
+| s16  | [-(2^15), 2^15-1] |
+| s32  | [-(2^31), 2^31-1] |
+| s64  | [-(2^63), 2^63-1] |
+| u8   | [0, 2^8-1]        |
+| u16  | [0, 2^16-1]       |
+| u32  | [0, 2^32-1]       |
+| u64  | [0, 2^64-1]       |
+
+These types differ from the core value types in that they represent a range of integers instead of a set of bits.
+
+## `string`
+
+A string value is a homogenous sequence of [unicode code points](https://www.unicode.org/glossary/#code_point).
```

Comment:
what does homogeneous mean here?

Reply:
It's possibly redundant. The intent was to specify that it's not a heterogeneous sequence, like a tuple.

Reply:
I think that the key concept is sequence. It is not an array of codepoints (thanks unicode) and it is definitely not a tuple.
At some point we are going to have sequences; in which case our story for strings should be layerable on top of the story for sequences. Even if we have special instructions for string handling; they should be re-interpret-able as macros over sequences of codepoints.

Reply:
Removed the homogeneous keyword.